agent/xds/listeners.go

```diff
@@ -1146,6 +1150,13 @@ func (s *ResourceGenerator) makeFilterChainTerminatingGateway(
 
 		opts.cluster = ""
 		opts.useRDS = true
+
+		if meshConfig := cfgSnap.MeshConfig(); meshConfig == nil || meshConfig.HTTP == nil || !meshConfig.HTTP.SanitizeXForwardedClientCert {
+			opts.forwardClientDetails = true
+			// Note: filter Connection may not be mTLS, so then ALWAYS_FORWARD_ONLY. For mTLS connections we might want APPEND_FORWARD.
+			// Open question; how do I determine if this is mTLS or not?
```

Comment:
These filter chains use mTLS, the TLS context set above requires client certificates:
https://github.com/hashicorp/consul/blob/3b3f001580ea919057478657902cd5f4aae378aa/agent/xds/listeners.go#L1097

I don't think that `ALWAYS_FORWARD_ONLY` would work for terminating gateways because the header would never be set.  

`APPEND_FORWARD` is currently only set on the _inbound_ listener, and terminating gateways would only be dialed by a connect proxy's _upstream_ listener. 

This might need to be `APPEND_FORWARD`.

Reply:
I agree, changed.